### DISCUSSION TITLE
TeslaMate not starting up -database issues

### DISCUSSION THREAD

**Discussion by MrMEScott** [teslamate-org/teslamate#1291] [Q&A]
I'm running the latest available TeslaMate via docker-compose on a Raspberry Pi 4B. It's worked flawlessly for many months but today it's refusing web connections. I stopped then restarted via docker-compose and I see the following in the log:

```
teslamate_1  | 2021-02-04 11:26:25.400 [error] Could not create schema migrations table. This error usually happens due to the following:
teslamate_1  | 
teslamate_1  |   * The database does not exist
teslamate_1  |   * The "schema_migrations" table, which Ecto uses for managing
teslamate_1  |     migrations, was defined by another library
teslamate_1  |   * There is a deadlock while migrating (such as using concurrent
teslamate_1  |     indexes with a migration_lock)
teslamate_1  | 
teslamate_1  | To fix the first issue, run "mix ecto.create".
teslamate_1  | 
teslamate_1  | To address the second, you can run "mix ecto.drop" followed by
teslamate_1  | "mix ecto.create". Alternatively you may configure Ecto to use
teslamate_1  | another table and/or repository for managing migrations:
teslamate_1  | 
teslamate_1  |     config :teslamate, TeslaMate.Repo,
teslamate_1  |       migration_source: "some_other_table_for_schema_migrations",
teslamate_1  |       migration_repo: AnotherRepoForSchemaMigrations
teslamate_1  | 
teslamate_1  | The full error report is shown below.
teslamate_1  | 
teslamate_1  | ** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 2941ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:
teslamate_1  | 
teslamate_1  |   1. Tracking down slow queries and making sure they are running fast enough
teslamate_1  |   2. Increasing the pool_size (albeit it increases resource consumption)
teslamate_1  |   3. Allowing requests to wait longer by increasing :queue_target and :queue_interval
teslamate_1  | 
teslamate_1  | See DBConnection.start_link/2 for more information
```
I am not enough of a geek to tackle any of those prescriptions - does anyone have any simple suggestions as to how I should proceed? I have a fairly recent backup of the database that I could restore.

**Comment by MrMEScott** [comment]
The log for the database service contains this:

```
database_1   | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1   | 
database_1   | 2021-02-04 12:04:30.942 UTC [1] LOG:  starting PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on arm-unknown-linux-gnueabihf, compiled by gcc (Debian 8.3.0-6) 8.3.0, 32-bit
database_1   | 2021-02-04 12:04:30.943 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1   | 2021-02-04 12:04:30.943 UTC [1] LOG:  listening on IPv6 address "::", port 5432
database_1   | 2021-02-04 12:04:30.948 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
database_1   | 2021-02-04 12:04:30.960 UTC [27] LOG:  database system was interrupted; last known up at 2021-02-03 00:37:32 UTC
database_1   | 2021-02-04 12:04:31.196 UTC [27] LOG:  invalid primary checkpoint record
database_1   | 2021-02-04 12:04:31.196 UTC [27] PANIC:  could not locate a valid checkpoint record
database_1   | 2021-02-04 12:04:32.885 UTC [1] LOG:  startup process (PID 27) was terminated by signal 6: Aborted
database_1   | 2021-02-04 12:04:32.885 UTC [1] LOG:  aborting startup due to startup process failure
database_1   | 2021-02-04 12:04:32.966 UTC [1] LOG:  database system is shut down
```
I guess this means I should restore the backup, however the [instructions for restoring](https://docs.teslamate.org/docs/maintenance/backup_restore/) it don't work because the database service isn't running. What now?

**Comment by tobiasehlert** [comment]
Hi @MrMEScott,

Have you resolved your issue?

Postgres is looking for a checkpoint record in the transaction log that probably doesn't exist or is corrupted. You can determine if this is the case by running following postgres command (since you are probably on postgres version 10 or higher):

```
pg_resetwal DATADIR
```

If the transaction log is corrupt, you'll see a message like:

> The database server was not shut down cleanly. Resetting the transaction log might cause data to be lost. If you want to proceed anyway, use `-f` to force reset.

You can then follow the instructions and run with -f to force the update for Postgres version 10 or higher:

```
pg_resetwal -f DATADIR
```

That should reset the transaction log, however it could leave your database in an indeterminate state.. so do a backup of your data before doing something like this.. not that your data gets lost totally. :)

Regards,
Tobias

**Reply by MrMEScott** [reply]
Thanks for following up. In the end I gave up on trying to restore the database and did a complete reinstall. I still have a backup of the old corrupt database so I guess I could try to extract at least some of the historical entries and merge these into the current db - but I don't really have time to spend on it right now.

**Reply by tobiasehlert** [reply]
Alright @MrMEScott!
Would maybe me a nice try to see, if it actually was the issue and if it resolves.

And importing yeah.. nice to have historical details, so hopefully you find some spare time soon so you can get your history back  =D
